msprime/demography.py

```diff
@@ -850,10 +850,19 @@ def insert_extra_populations(self, tables):
                     md["name"] = population.name
                 if description_in_metadata:
                     md["description"] = population.description
-                # TODO we could also try to merge in the ``extra_metadata``
```

Comment:
We should keep the TODO I think, it's not quite the same thing as you're doing here. Adding in stuff from extra_metadata if it's present in ``properties`` is definitely the right thing to do.

Reply:
Ah, right. I've gone ahead and done the TODO.